ci/tests/arakoon/extend_promote_demote_test.py

```diff
@@ -25,24 +25,182 @@
 # promote will extend cluster / demote will reduce cluster
 #
 
+from ovs.extensions.db.arakoon.ArakoonInstaller import ArakoonInstaller, ArakoonClusterConfig
+from ovs.extensions.generic.sshclient import SSHClient
 from ovs.lib.storagedriver import StorageDriverController
+from ovs.lib.albacontroller import AlbaController
 
 from ci.tests.backend import alba, generic
 from ci.tests.disklayout import disklayout
-from ci.tests.general import general_ovs
 from ci.tests.general import general
+from ci.tests.general import general_ovs
+
 from nose.plugins.skip import SkipTest
+from nose.tools import assert_raises, assert_false, assert_true
+
+from ci.tests.general.logHandler import LogHandler
+
+from ConfigParser import RawConfigParser
+
+import os
+from StringIO import StringIO
+
+
+logger = LogHandler.get('arakoon', name='setup')
+logger.logger.propagate = False
+
+BACKEND_NAME = 'OVS_3554'
+BACKEND_TYPE = 'alba'
+PMACHINES = general_ovs.get_pmachines_by_ip()
+
+
+def are_files_present_on(client, files):
+    for filename in files:
+        assert client.file_exists(filename) == True, 'File {0} not present'.format(filename)
+
+
+def are_files_missing_on(client, files):
+    for filename in files:
+        assert client.file_exists(filename) == False, 'File {0} is missing'.format(filename)
+
+
+def get_cluster_pmachines(ips):
+    pmachines_to_check = dict()
+    for ip in ips:
+        pmachines_to_check[ip] = PMACHINES.get(ip)
+    return pmachines_to_check
 
 
 def setup():
-    pass
+    logger.info('setup alba backend')
+    pmachines = general_ovs.get_pmachines_by_ip()
+    master_ips = [ip for ip in pmachines.keys() if pmachines[ip]['node_type'] == 'MASTER']
+    master_ips.sort()
+
+    for ip in pmachines.keys():
+        storagerouter = general_ovs.get_storagerouter_by_ip(ip)
+        disklayout.add_db_role(storagerouter['guid'])
+
+    if generic.is_backend_present(BACKEND_NAME, BACKEND_TYPE):
+        backend = generic.get_backend_by_name_and_type(BACKEND_NAME, BACKEND_TYPE)
+        alba.remove_alba_backend(backend['alba_backend_guid'])
+
+    _ = alba.add_alba_backend(BACKEND_NAME)
+    backend = generic.get_backend_by_name_and_type(BACKEND_NAME, BACKEND_TYPE)
+
+    logger.info('running voldrv arakoon checkup ...')
+    StorageDriverController.manual_voldrv_arakoon_checkup()
+    logger.info('running alba arakoon checkup ...')
+    AlbaController.manual_alba_arakoon_checkup(backend['alba_backend_guid'])
+    logger.info('running alba nsm checkup ...')
+    AlbaController.nsm_checkup()
+
```

Comment:
The nsm checkup and alba arakoon checkup are not required here, as they have been executed during backend creation